microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -347,9 +347,14 @@ def generate_testpmd_command(
     def run_for_n_seconds(self, cmd: str, timeout: int) -> str:
         self._last_run_timeout = timeout
         self.node.log.info(f"{self.node.name} running: {cmd}")
-
+        # add delays such that receiver will always start first and end last
```

Comment:
For start first, you can use wait_output to wait for a more reliable signal from its output. The end last can use wait_result to wait for other processes end firstly. It doesn't need to hard code a sleep.

Reply:
This is going to require a much larger refactor